dvclive/__init__.py

```diff
@@ -35,9 +28,21 @@ def log(name: str, val: Union[int, float], step: int = None) -> None:
     if not _metric_logger:
         _metric_logger = MetricLogger()
 
+    return _metric_logger
+
+
+def log(name: str, val: Union[int, float], step: int = None) -> None:
+    global _metric_logger  # pylint: disable=global-statement
+    _metric_logger = _lazy_init(_metric_logger)
     _metric_logger.log(name=name, val=val, step=step)
 
 
+def get_step() -> None:
+    global _metric_logger  # pylint: disable=global-statement
+    _metric_logger = _lazy_init(_metric_logger)
+    return _metric_logger.step
```

Comment:
This method returns int.
Also, I think we shoul change `MetricLogger`'s `step` property into `get_step()` method to maintain consistency with with API. @daavoo  what do you think?

Reply:
Not sure, tbh. `get_X` method instead of `@property` kind of feels strange and `step` doesn't sound good for a public method neither 🤷

Reply:
Well, my POV is that I presume that at some point one might want to, parallelize their code, and in that case do something like:
`dvclive = MetricsLogger()` in that case, `dvclive.get_step` stops working.
Now that I mention that, it would probably be good to mention that dvclive is not thread-safe, and one needs to initialize their own Loggers in case of parallel jobs.

Reply:
I see your point. 

However, assuming we are focusing on "integrations first" (https://github.com/iterative/example-repos-dev/pull/77#issuecomment-912419279), the parallelization would happen at the ML Framework level and most ML Frameworks already take care of properly calling the *callbacks*/*loggers*.

Reply:
What's the downside to having the public API match `MetricsLogger`? 

I don't think I understand the point about ML framework integrations. Even if ML frameworks spawn a separate process for each model training, dvclive would try to read/write using the same file by default, right? Users might need to specify a different path for each one, which isn't supported yet in the callbacks.

Reply:
I would need to investigate each case but, at least in the Deep Learning Frameworks I'm familiar with, parallelism usually occurs:

- At the Data Loader level 
Which doesn't affect DVCLive callbacks.

- In Distributed training strategy
Where ML Framework usually provide some decorator like [rank_zero_only](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/utilities/distributed.py#L45) / [master_only]( https://github.com/open-mmlab/mmcv/blob/c47c9196d067a0900b7b8987a8e82768edab2fff/mmcv/runner/dist_utils.py#L98 ) which is (should be) used in the DVCLive callback.